src/services/smart-contracts/actions/core.js

```diff
@@ -106,7 +111,134 @@ function getReadyCampaign(campaign, identity, Dai) {
 	return newCampaign
 }
 
-export async function openChannel({ campaign, account }) {
+async function getChannelsWithBalance({ identityAddr, requester, STATES }) {
+	const channels = await requester
+		.fetch({
+			route: '/campaigns',
```

Comment:
you should use `?byEarner=` here

Reply:
that shouldn't be here at all, there is market service for that calls